### PR TITLE
More tweaks for notification logic

### PR DESCRIPTION
**Pull request by infinity0**
TL;DR is that the previous behaviour was mostly fine. one just has to look for `notify_contact` in arg2 rather than `_notifications` in arg1. Example trace running without any `fd-gui`:

~~~~
disconnected disconnect
connected connect
connected_notifications notify_contact                  # message received on inactive contact
connected_notifications notify_contact
connected clear_all_notifications                       # visited all contacts
connected notify_contact clear_all_notifications        # message received on active contact
connected notify_contact clear_all_notifications        # and again
                                                        # now switching to a different contact
connected_notifications notify_contact                  # message received on previous contact, now inactive
connected_notifications notify_contact
connected clear_all_notifications                       # visited all contacts again
~~~~

More generally, if `fd-gui` is not set then jackline should behave assuming the parent gui is always in focus - I initialise `gui_has_focus` to `true` and no other place sets it to `false`.

I also fixed a few other minor things, more details in the commit messages.

### CODE REVIEW

File: bin/jackline.ml [hunk 1]
```diff
@@ -93,8 +93,8 @@ let start_client cfgdir debug unicode fd_gui fd_nfy () =
 
   let ui_mvar = Lwt_mvar.create_empty () in
 
-  Cli_state.Notify.notify_writer myjid config.Xconfig.notification_callback fd_nfy >>= fun state_mvar ->
-  Cli_state.Notify.gui_focus_reader fd_gui ui_mvar ;
+  Cli_state.Notify.notify_writer ui_mvar myjid config.Xconfig.notification_callback fd_nfy >>=
+  fun state_mvar -> Cli_state.Notify.gui_focus_reader fd_gui ui_mvar ;
```
Comment (hannesm) [inline bin/jackline.ml]: your linewrapping here is different than the rest of this project

File: cli/cli_state.ml [hunk 2]
```diff
@@ -130,8 +132,8 @@ module Notify = struct
     Lwt_unix.openfile fd [flag_of_mode mode] 0 >|=
     Lwt_io.of_fd ~mode:mode
 
-  (* TODO(infinity0): perhaps log the exception somewhere *)
-  let catch_exn f = Lwt.catch f (fun _ -> Lwt.return_unit)
+  let catch_exn ui_mvar f = Lwt.catch f (fun e ->
+      selflog ui_mvar "async error in notify r/w" (Printexc.to_string e))
```
Comment (hannesm) [inline cli/cli_state.ml]: does at no point the `state_mvar` get pushed by the loop reading `ui_mvar` (if this is the case, there is a potential deadlock (see https://github.com/hannesm/jackline/commit/90b8068adf218fd73e87f33c3cf57417bacdf0ec for a solution of a similar issue))

**Comment by hannesm** [comment]
thanks for the explanation, I merged https://github.com/hannesm/jackline/pull/134/commits/624ba7be16edca5eaacca0c93728b61c39951a03 and https://github.com/hannesm/jackline/pull/134/commits/76cdd5736f01456a98f42ad1ebc277bd9d5bbe43 into master... feel free to open another PR for a polished https://github.com/hannesm/jackline/pull/134/commits/bfa3877b5f62b15c7f9a50c7125e7d46a2f48ebd
